python/tchannel/tornado/message_factory.py

```diff
@@ -36,6 +41,156 @@ def __init__(self):
+
+        :param request: Request
+        :param args: array of arg streams
+        :param is_completed: message flags
+        :return: CallRequestMessage/CallRequestContinueMessage
+        """
+        request.flags = FlagsType.none if is_completed else FlagsType.fragment
+
+        # TODO decide what need to pass from request
+        if request.state == StreamState.init:
+            message = CallRequestMessage(
+                flags=request.flags,
+                # ttl=request.ttl,
+                # tracing=request.tracing,
+                service=request.service,
+                # headers=request.headers,
+                # checksum=request.checksum
+                args=args
+            )
+            request.state = (StreamState.completed if is_completed
+                             else StreamState.streaming)
+        elif request.state == StreamState.streaming:
+            message = CallRequestContinueMessage(
+                flags=request.flags,
+                # checksum=request.checksum
+                args=args
+            )
+            request.state = (StreamState.completed if is_completed
+                             else StreamState.streaming)
+        else:
+            raise StreamingException("request state Error")
+
+        return message
+
+    def build_raw_response_message(self, response, args, is_completed=False):
+        """build protocol level message based on response and args.
+
+        response object contains meta information about outgoing response.
+        args are the currently chunk data from argstreams
+        is_completed tells the flags of the message
+
+        :param response: Response
+        :param args: array of arg streams
+        :param is_completed: message flags
+        :return: CallResponseMessage/CallResponseContinueMessage
+        """
+        response.flags = FlagsType.none if is_completed else FlagsType.fragment
+
+        # TODO decide what need to pass from request
+        if response.state == StreamState.init:
+            message = CallResponseMessage(
+                flags=response.flags,
+                # code=response.code,
+                # tracing=response.tracing,
+                # headers=response.headers,
+                # checksum=response.checksum
+                args=args
+            )
+            response.state = (StreamState.completed if is_completed
+                              else StreamState.streaming)
+        elif response.state == StreamState.streaming:
+            message = CallResponseContinueMessage(
+                flags=response.flags,
+                # checksum=response.checksum
+                args=args
+            )
+            response.state = (StreamState.completed if is_completed
+                              else StreamState.streaming)
+        else:
+            raise StreamingException("response state Error")
+
+        return message
+
+    def build_raw_message(self, context, args, is_completed=False):
```

Comment:
`context` implies it's a `Context` from context.py.

If not, use a different name.